cupy/indexing/generate.py

```diff
@@ -1,9 +1,144 @@
 # flake8: NOQA
 # "flake8: NOQA" to suppress warning "H104  File contains nothing but comments"
+import cupy
+
+import six
+
+
+class AxisConcatenator(object):
+    """Translates slice objects to concatenation along an axis.
+
+    For detailed documentation on usage, see `r_`.
+
+    """
+    def _output_obj(self, newobj, tempobj, ndmin, trans1d):
+        k2 = ndmin - tempobj.ndim
+        if (trans1d < 0):
+            trans1d += k2 + 1
+        defaxes = list(six.moves.range(ndmin))
+        k1 = trans1d
+        axes = defaxes[:k1] + defaxes[k2:] + \
+               defaxes[k1:k2]
+        newobj = newobj.transpose(axes)
+        return newobj
+
+    def __init__(self, axis=0, matrix=False, ndmin=1, trans1d=-1):
+        self.axis = axis
+        self.trans1d = trans1d
+        self.matrix = matrix
+        self.ndmin = ndmin
+
+    def __getitem__(self, key):
+        trans1d = self.trans1d
+        ndmin = self.ndmin
+        objs = []
+        if isinstance(key, str):
```

Comment:
It cannot catch the case when key is `unicode` type like `a[u'foo']` when we use Python 2. If we do want to exclude the case we should use `isinstance(key, six.string_types)` instead.